lib/rules/complexity.js

```diff
@@ -61,12 +64,16 @@ module.exports = {
     create(context) {
         const option = context.options[0];
         let THRESHOLD = 20;
+        let IS_MODIFIED_COMPLEXITY = false;
+
+        if (typeof option === "object") {
+            if (Object.hasOwn(option, "maximum") || Object.hasOwn(option, "max")) {
+                THRESHOLD = option.maximum || option.max;
+            }
```

Comment:
```suggestion
        let IS_MODIFIED_COMPLEXITY = false;

        if (typeof option === "object") {
            THRESHOLD = option?.maximum ?? option?.max ?? THRESHOLD;
            IS_MODIFIED_COMPLEXITY = option?.modifiedComplexity ?? IS_MODIFIED_COMPLEXITY;
        }
```

Reply:
This would change the behavior of the rule because now you are explicitly looking for `undefined` rather than just ORing values together.

I also think the original is easier to understand.